src/KingSystem/System/Hio.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <heap/seadDisposer.h>
+#include <hostio/seadHostIONode.h>
+#include "KingSystem/Utils/Types.h"
+
+namespace ksys {
+
+class Hio final : public sead::hostio::Node {
+    SEAD_SINGLETON_DISPOSER(Hio)
+
+public:
```

Comment:
Constructors/destructors for singleton classes should be private so that nobody is tempted to create a second instance of a singleton.